### ISSUE TITLE
How to predict with a lda2vec model

### ISSUE THREAD

**Issue by alexwongdl** [cemoody/lda2vec#52]
When I have a lda2vec model, how can I predit the vector of a document with this model. I couldn't find the prediction method in any of following files : corpus.py  dirichlet_likelihood.py  embed_mixture.py  fake_data.py  __init__.py  negative_sampling.py  preprocess.py  topics.py  tracking.py  utils.py.

**Comment by yg37** [comment]
It doesn't have a prediction function yet

**Comment by ali3assi** [comment]
@AlexWang90 @yg37 did you have a solution for predicting the embedding for a new document?
can you please share your slolution

**Comment by yg37** [comment]
@TamouzeAssi  I wrote a prediction function and tested it on some new documents. The results are not very good (I think it's because I did not run enough epochs). I can send you the function if you want to experiment with it more.

**Comment by ali3assi** [comment]
@yg37 yes please send it, i wanna try it and check what can i do and improve it. thank you

**Comment by yg37** [comment]
@TamouzeAssi What's your email address?

**Comment by TamouzeAssi** [comment]
@yg37 ali3assi@gmail.com

**Comment by nateraw** [comment]
@yg37 can you email me? I tried to reach out via the email on your bio, but it seems it may be dead! My email is nxr9266@rit.edu . Thank you!

**Comment by yg37** [comment]
Hi Nathan,

Sorry about that - the school is only valid for a year after I graduate and
I forgot to edit it.

The attached are the two files:
lda2vec_server.py for serving the models
lda2vec_model.py contains the actual prediction models

I haven't tested it out too much and it takes a lot of epochs for the
prediction results to make sense, so it does not quite make sense to serve
it in real time. But I would love to hear about how it works out for you!

Best,
Sophie

**Comment by nateraw** [comment]
@yg37 Thank you for your quick reply!! I am using Tensorflow, as I don't know Chainer. Can you give me a theoretical understanding of what you are doing, please? 

This is my current thought process:

1) Append a new random document weight to document matrix (Within the range of current weights)
2) Collect skipgram pairs for this document (~30-100 pairs)
3) Collect random sampling of previously trained skipgram pairs (~10000 pairs)
4) Shuffle the two lists of pairs together
5) Train using the combined list of skipgram pairs, locking word vectors
6) Evaluate the new document/topic embeddings

Am I close to what you are doing? Again, thank you!!

**Comment by arhelgens** [comment]
@yg37 I am I would also be interested in seeing what you have implemented for a prediction function. 

Any way that you could either publish your code, or send me what you have. 

Thanks

**Comment by simon19891101** [comment]
Hi @yg37 just wondering if I could have the implementation of your prediction function? My email is simon19891101@gmail.com. Much appreaciated :).

**Comment by raziel130889** [comment]
Hi @yg37  I would like to use your function to predict, please could you send it to me, this is my email raziel130889@gmail.com, in advance thank you

**Comment by sudheernaidu53** [comment]
hi, 
@yg37 ,I am working with lda2vec as well, I will be thankful if you can send the predict function across to me.

**Comment by Shivanshu156** [comment]
Please send the predict function here as well
shivanshu156@gmail.com

**Comment by kandakji** [comment]
Hi @yg37 I would much appreciate it if you would send me your predict function implementation :) 
kondakji.oussama@gmail.com
Thank you in advance. :)

**Comment by leewide** [comment]
Hi @yg37 I would much appreciate it if you would send me your predict function implementation, this is my email lihuaide195536@gmail.com, in advance thank you
